### PR TITLE
I18n/select locale

### PR DESCRIPTION
**Pull request by petersg83**
I let `created_by` and `updated_by` in the model as it may be useful and doesn't bring drawbacks.

### CODE REVIEW

File: packages/strapi-plugin-content-manager/services/data-mapper.js [hunk 1]
```diff
@@ -98,6 +98,7 @@ const HIDDEN_CONTENT_TYPES = [
   'plugins::upload.file',
   'plugins::users-permissions.permission',
   'plugins::users-permissions.role',
+  'plugins::i18n.locale',
```
Comment (alexandrebodin) [inline packages/strapi-plugin-content-manager/services/data-mapper.js:101]: We need to introduce something so this becomes dynamic we can't let that be static anymore

File: packages/strapi-plugin-i18n/services/locales.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+const find = async (...args) => {
```
Comment (alexandrebodin) [inline packages/strapi-plugin-i18n/services/locales.js:3]: looks like you can return the query call everytime here

File: packages/strapi-plugin-i18n/tests/locales.test.e2e.js [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        url: '/i18n/locales',
+        method: 'POST',
+        body: locale,
+      });
+
+      expect(res.statusCode).toBe(400);
+      expect(res.body).toMatchObject({ message: 'This locale already exists' });
+    });
+
+    // - Faire documentation
```
Comment (alexandrebodin) [inline packages/strapi-plugin-i18n/tests/locales.test.e2e.js:105]: :D

File: test/helpers/test-app-generator.js [hunk 4]
```diff
@@ -54,6 +54,7 @@ const generateTestApp = async ({ appName, database }) => {
       'strapi-plugin-graphql',
       'strapi-plugin-i18n',
```
Comment (alexandrebodin) [inline test/helpers/test-app-generator.js:55]: didn't create a conflict but it's duplicated :)

File: packages/strapi-plugin-i18n/controllers/locales.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const { setCreatorFields, sanitizeEntity } = require('strapi-utils');
+const { getService } = require('../utils');
+const { validateCreateLocaleInput, validateUpdateLocaleInput } = require('../validation/locales');
+const { formatLocale } = require('../domain/locale');
+
+const listLocales = async ctx => {
```
Comment (alexandrebodin) [inline packages/strapi-plugin-i18n/controllers/locales.js:8]: I would create a sanitizeLocale fn so we don't need to getModel et call sanitizeEntity in each handler

**Review by alexandrebodin — CHANGES_REQUESTED** [review]
First pass a few notes here and there.

File: packages/strapi-plugin-content-manager/services/data-mapper.js (hunk 1 above)
Comment (petersg83) [reply packages/strapi-plugin-content-manager/services/data-mapper.js:101]: What do you suggest?
An option in the models ? (I prefer this one)
A service where plugin can register the hidden models ?

Comment (alexandrebodin) [reply packages/strapi-plugin-content-manager/services/data-mapper.js:101]: An option in the models is fine. Can you write up an ADR to document the existing options & propose a new or updated one so we can take a decision :)

Comment (petersg83) [reply packages/strapi-plugin-content-manager/services/data-mapper.js:101]: I'll do it in another PR :)

**Review by alexandrebodin — CHANGES_REQUESTED** [review]
make sure to export the controllers

```js
module.exports = {
  actionA() {},
  actionB() {},
  actionC() {},  
}
```

File: packages/strapi-plugin-i18n/oas.yml [hunk 6]
```diff
@@ -0,0 +1,141 @@
+openapi: 3.0.2
```
[the remark this answers is not in the document]
Comment (petersg83) [reply packages/strapi-plugin-i18n/oas.yml:1]: ☺️

**Review by alexandrebodin — APPROVED** [review]
LGTM

**Pull request merged**